server/src/main/java/org/opensearch/threadpool/ThreadPool.java

```diff
@@ -383,19 +383,21 @@ public ThreadPoolStats stats() {
             long rejected = -1;
             int largest = -1;
             long completed = -1;
-            if (holder.executor() instanceof ThreadPoolExecutor) {
-                ThreadPoolExecutor threadPoolExecutor = (ThreadPoolExecutor) holder.executor();
+            long waitTime = -1;
+            if (holder.executor() instanceof OpenSearchThreadPoolExecutor) {
```

Comment:
We should keep the previous condition as is and just fetch the `waitTime` in case the executor type is `OpenSearchThreadPoolExecutor` otherwise default to -1. There can be some executors (probably provided by plugins?) which can be of type other than OpenSearchThreadPoolExecutor and we will miss existing stats for such cases.

Reply:
It looks like the `ExecutorHolder` class already has an `assert` for `OpenSearchThreadPoolExecutor` so I think this should be fine actually
https://github.com/opensearch-project/OpenSearch/blob/742267dba8599b60bb5bb16bb69ad4ba114a5581/server/src/main/java/org/opensearch/threadpool/ThreadPool.java#L706-L719

The `DIRECT_EXECUTOR` is used for the "same" thread pool that we already don't gather stats on anyways
https://github.com/opensearch-project/OpenSearch/blob/bbe6b2d85430f2aeffd91b8950262d75eb7a76e4/server/src/main/java/org/opensearch/threadpool/ThreadPool.java#L300

 From the [javadoc](https://github.com/jed326/OpenSearch/blob/4ad418210a51c518119a4c9c565fbf7e9bc4b5c1/server/src/main/java/org/opensearch/common/util/concurrent/OpenSearchThreadPoolExecutor.java#L44-L49) `OpenSearchThreadPoolExecutor` is intended for additional stats collection so to me it seems like this change should be done sooner or later.